File: 백준/Silver/25206. 너의 평점은/너의 평점은.c
#include <stdio.h>
#include <string.h>

int main(void) {
	char name[51];
	float score;
	char grade[3];
	float total, avg = 0, num = 0;
	for (int i = 0; i < 20; i++) {
		scanf("%s %f %s", name, &score, grade);
		if (grade[0] == 'P') continue;
		num += score;
		if (grade[0] == 'A') total = 4;
		else if (grade[0] == 'B') total = 3;
		else if (grade[0] == 'C') total = 2;
		else if (grade[0] == 'D') total = 1;
		else if (grade[0] == 'F') continue;
		if (grade[1] == '+') total += 0.5;
		
		avg += score * total;
	}						 
	printf("%f", avg / num);
							 
	return 0;				 
}	

File: 백준/Silver/25206. 너의 평점은/너의 평점은.py
total = 0
hap = 0
avg = 0


for i in range(20):
    name, score, rate = input().split()
    name = list(name)[:50]
    score = float(score)
    rate = list(rate)

    if(rate[0] == 'P'): continue
    total += score
    if(rate[0] == 'A'): hap = 4
    elif(rate[0] == 'B'): hap = 3
    elif(rate[0] == 'C'): hap = 2
    elif(rate[0] == 'D'): hap = 1
    elif(rate[0] == 'F'): continue
    if(rate[1] == '+'): hap += 0.5

    avg += score * hap
print("%f" %(avg/total))